server/middleware/authz.go

```diff
@@ -0,0 +1,378 @@
+		api.CreateBranchMethodName,
+		api.DeleteBranchMethodName,
+		api.ListBranchesMethodName,
+		api.CreateAppKeyMethodName,
+		api.UpdateAppKeyMethodName,
+		api.DeleteAppKeyMethodName,
+		api.ListAppKeysMethodName,
+		api.RotateAppKeySecretMethodName,
+
+		// auth
+		api.GetAccessTokenMethodName,
+		api.CreateInvitationsMethodName,
+		api.DeleteInvitationsMethodName,
+		api.ListInvitationsMethodName,
+		api.VerifyInvitationMethodName,
+		api.ListUsersMethodName,
+
+		// billing
+		api.ListInvitationsMethodName,
+
+		// cache
+		api.CreateCacheMethodName,
+		api.ListCachesMethodName,
+		api.DeleteCacheMethodName,
+		api.SetMethodName,
+		api.GetSetMethodName,
+		api.GetMethodName,
+		api.DelMethodName,
+		api.KeysMethodName,
+
+		// health
+		api.HealthMethodName,
+
+		// management
+		api.InsertUserMetadataMethodName,
+		api.GetUserMetadataMethodName,
+		api.UpdateUserMetadataMethodName,
+		api.InsertNamespaceMetadataMethodName,
+		api.GetNamespaceMetadataMethodName,
+		api.UpdateNamespaceMetadataMethodName,
+		api.ListNamespacesMethodName,
+		api.CreateNamespaceMethodName,
+		api.DeleteNamespaceMethodName,
+
+		// observability
+		api.QueryTimeSeriesMetricsMethodName,
+		api.QuotaLimitsMetricsMethodName,
+
+		// realtime
+		api.PresenceMethodName,
+		api.GetRTChannelMethodName,
+		api.GetRTChannelsMethodName,
+		api.ReadMessagesMethodName,
+		api.MessagesMethodName,
+		api.ListSubscriptionsMethodName,
+	)
+)
+
+func authzUnaryServerInterceptor() func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		if config.DefaultConfig.Auth.Authz.Enabled {
+			reqMetadata, _ := request.GetRequestMetadataFromContext(ctx)
+			role := getRole(reqMetadata)
+
+			// empty role check for transition purpose
+			if role != "" {
+				authorized := isAuthorized(reqMetadata.GetFullMethod(), role)
+				if !authorized {
+					return nil, errors.PermissionDenied(fmt.Sprintf("You are not allowed to perform operation: %s", reqMetadata.GetFullMethod()))
+				}
+			}
+			resp, err := handler(ctx, req)
+			return resp, err
+		}
+		return handler(ctx, req)
+	}
+}
+
+func authzStreamServerInterceptor() grpc.StreamServerInterceptor {
+	return func(srv interface{}, stream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+		reqMetadata, _ := request.GetRequestMetadataFromContext(stream.Context())
+		role := getRole(reqMetadata)
+		// empty role check for transition purpose
+		if role != "" {
+			authorized := isAuthorized(reqMetadata.GetFullMethod(), role)
+			if !authorized {
+				return errors.PermissionDenied(fmt.Sprintf("You are not allowed to perform operation: %s", reqMetadata.GetFullMethod()))
+			}
+		}
+		return handler(srv, stream)
+	}
+}
+
+func isAuthorized(methodName string, role string) bool {
+	allowed := false
+	switch role {
```

Comment:
```
allowed := false
if methods := getMethodsForRole(role); methods != nil {
  allowed = methods.Contains(methods)
}
```
and move the switch to getMethodsForRole.

Reply:
sure updated.